nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/ecm/admin/operation/HotReloadStudioSnapshot.java

```diff
@@ -71,46 +80,110 @@
     protected boolean validate = true;
 
     @OperationMethod
-    public void run() throws Exception {
+    public Blob run() {
+
+        JSONArray result = new JSONArray();
+        JSONObject resultJSON = new JSONObject();
+
         if (updateInProgress) {
-            return;
+            resultJSON.put("status", "inProgress");
```

Comment:
You should declare all statuses as constant in this class for example. As it we can have a quick look to know which statuses we could have.

Reply:
You should also use a helper function that takes a `String status, String message` and returns the build `JSONBlob`.